utils/upgrade

```diff
@@ -0,0 +1,345 @@
+                return Failure(f"Compilation of Open CAS failed. Build log: {COMPILATION_LOG}")
+
+        return Success()
+
+
+class InstallCas(UpgradeState):
+    log = "Installing new Open CAS files"
+
+    def do_work(self):
+        with open(COMPILATION_LOG, "a") as install_log:
+            p = subprocess.run(
+                ["make", "install"], cwd="..", stdout=install_log, stderr=install_log
+            )
+
+            if p.returncode:
+                return Failure(
+                    f"Installation of Open CAS failed. Installation log: {COMPILATION_LOG}"
+                )
+
+        return Success()
+
+
+class InsertModule(UpgradeState):
+    module_path = f"{os.path.dirname(__file__)}/../modules/cas_cache/cas_cache.ko"
+    options = {"installed": False}
+
+    def do_work(self):
+        try:
+            insert_module(self.module_path, **self.options)
+        except Exception as e:
+            return Failure(f"Couldn't load module {self.module_path}. Reason: {e}")
+
+        return Success()
+
+
+class InsertNewModule(InsertModule):
+    log = "Try to insert new caching module"
+
+
+class DryRun(InsertModule):
+    log = "Perform dry run to check upgrade data integrity"
+    options = {"installed": False, "dry_run": 1}
+
+    def do_work(self):
+        result = super().do_work()
+        if not isinstance(result, Success):
+            return Failure("Dry run failed")
+
+        try:
+            logging.info(
+                "Insert with dry_run set succeeded. Removing module to attempt final insertion."
+            )
+            remove_module("cas_cache")
+        except Exception as e:
+            return Failure("Could not unload module. Reason: {e}")
+
+        return result
+
+
+class InsertInstalledModule(InsertModule):
+    log = "Restore old cas_cache module"
+    module_path = "cas_cache"
+    options = {"installed": True}
+
+
+class PrepareForUpgrade(UpgradeState):
+    log = "Preparing Open CAS for upgrade"
+
+    def do_work(self):
+        try:
+            logging.info("Switching CAS to upgrade mode")
+            opencas.casadm.start_upgrade()
+        except opencas.casadm.CasadmError as e:
+            return Failure(e)
+
+        return Success()
+
+
+class RemoveModule(UpgradeState):
+    log = "Removing cas_cache module"
+
+    def do_work(self):
+        try:
+            logging.info("Removing cas_cache module")
+            remove_module("cas_cache")
+        except Exception as e:
+            return Failure(f"Failed to remove cas_cache module. Reason: {e}")
+
+        return Success()
+
+
+class SetSchedulersToNoop(UpgradeState):
+    log = "Setting core devices schedulers to noop"
+
+    def do_work(self):
+        unique_core_sysfs_devices = {
+            get_device_sysfs_path(core["device"])
+            for _, core in opencas.get_devices_state()["cores"].items()
+        }
+
+        self.state_machine.schedulers = {}
+        for core in unique_core_sysfs_devices:
```

Comment:
Maybe change core to core_sysfs_path? For me core is not self-explainable enough.

Reply:
Done